### PR TITLE
chore: enable type checking

### PR DESCRIPTION
**Pull request by nathanielwarner**
### 🔗 Linked issue

https://energysage.atlassian.net/browse/CED-1931

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Enables type checking and adds it to CI. Migrates all existing JavaScript in Vue code to TypeScript. Some of the type issues are complex to solve, and these have just been ignored for now. A few actual misuses of variables were uncovered and fixed though.

### 🥼 Testing

Verified the functionality of components that were changed at all. Verified type checking now has no errors reported.

#### 🧐 Feedback Requested / Focus Areas

- General

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.
- [x] I have documented testing approach

### CODE REVIEW

File: es-ds-components/components/es-zip-code-form.vue [hunk 1]
```diff
@@ -85,13 +85,13 @@ const rules = {
 
 const { v$, validateState } = useEsForms(rules, state);
 
-const ctaForm = ref<HTMLFormElement>('ctaForm');
+const ctaForm = useTemplateRef('ctaForm');
```
Comment (tomleo) [inline es-ds-components/components/es-zip-code-form.vue:88]: that previous `HTMLFormElement` type comes in handy with upgrading

File: es-ds-components/utils/mime-type-finder.ts [hunk 2]
```diff
@@ -39,10 +39,10 @@ const mimeTypesAndHexes = [
 
 export const mimeTypes = mimeTypesAndHexes.map(({ mime }) => mime);
 
-export function getHumanReadableTypeFromMime(mimeType) {
+export function getHumanReadableTypeFromMime(mimeType: any) {
     return mimeTypesAndHexes.find((mimeTypeAndHex) => mimeTypeAndHex.mime === mimeType)?.humanReadable;
 }
 
-export default function findMimeType(hex) {
```
Comment (tomleo) [inline es-ds-components/utils/mime-type-finder.ts:46]: Do you know if this is being imported anywhere using the default export?

File: es-ds-docs/pages/molecules/file-input.vue [hunk 3]
```diff
@@ -94,50 +94,50 @@ onMounted(async () => {
...
-const events = ref([]);
+let fileObjects: File[] = [];
+const uploadUrls: Ref<{ name: string; uploadUrl: string }[]> = ref([]);
+const events: Ref<{ msg: string; variant: string; show: boolean }[]> = ref([]);
 
 const onSubmit = () => {
     uploadUrls.value = fileObjects.map(({ name }) => ({
         name,
-        uploadUrl: url,
+        uploadUrl: url.value,
```
Comment (tomleo) [inline es-ds-docs/pages/molecules/file-input.vue:104]: 👍

**Review by tomleo — APPROVED** [review]
👍

File: es-ds-components/utils/mime-type-finder.ts (hunk 2 above)
Comment (nathanielwarner) [reply es-ds-components/utils/mime-type-finder.ts:46]: It's being used [in es-file-input](https://github.com/EnergySage/es-ds/blob/bb6141913fd3d399f795a4ee45d344c4f9ae6ecc/es-ds-components/components/es-file-input.vue#L109). Prior to this change it was manually imported in that file.

File: es-ds-components/components/es-zip-code-form.vue (hunk 1 above)
Comment (nathanielwarner) [reply es-ds-components/components/es-zip-code-form.vue]: I'm actually not even sure how or if it was working before, given that [template refs](https://vuejs.org/guide/essentials/template-refs) are invoked using `useTemplateRef`, and [ref](https://vuejs.org/guide/essentials/reactivity-fundamentals#declaring-reactive-state-1) is something else entirely.

I've added back the `<HTMLFormElement>` though.

**Pull request merged**